Python/Product/VSInterpreters/Interpreter/CondaEnvironmentFactoryProvider.cs

```diff
@@ -60,6 +62,27 @@ class CondaEnvironmentFactoryProvider : IPythonInterpreterFactoryProvider {
             _globalProvider = globalProvider;
         }
 
+        public void Dispose() {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        ~CondaEnvironmentFactoryProvider() {
+            Dispose(false);
+        }
+
+        protected virtual void Dispose(bool disposing) {
+            if (!_isDisposed) {
+                _isDisposed = true;
+                if (_envsTxtWatcher != null) {
```

Comment:
`_envsTxtWatcherTimer` and `_envsTxtWatcher` are accessed both inside and outside lock statement. Is that ok?

Reply:
Didn't see the lock. No, it's not okay, I'll fix.